recipes/objectbox-generator/all/conanfile.py

```diff
@@ -1,48 +1,60 @@
-from conans import CMake, ConanFile, tools
-from conans.errors import ConanInvalidConfiguration
 import os
 
-required_conan_version = ">=1.33.0"
+from conan import ConanFile
+from conan.errors import ConanInvalidConfiguration
+from conan.tools.files import copy, download, get
+from conan.tools.layout import basic_layout
+
+required_conan_version = ">=1.47.0"
+
 
 class PackageConan(ConanFile):
     name = "objectbox-generator"
     description = "ObjectBox Generator based on FlatBuffers schema files (fbs) for C and C++"
     license = "GPL-3.0"
     url = "https://github.com/conan-io/conan-center-index"
     homepage = "https://github.com/objectbox/objectbox-generator"
-    topics = ("database", "code-generator", "objectbox")
+    topics = ("database", "code-generator", "objectbox", "pre-built")
+
+    package_type = "application"
     settings = "os", "arch", "compiler", "build_type"
+    no_copy_source = True
 
-    @property
-    def _source_subfolder(self):
-        return "source_subfolder"
-
-    def validate(self):
-        if self.settings.os not in ["Linux", "Windows", "Macos"] or self.settings.arch != "x86_64":
-            raise ConanInvalidConfiguration("{} doesn't support current environment".format(self.name))
+    def layout(self):
+        basic_layout(self, src_folder="src")
 
     def package_id(self):
         del self.info.settings.compiler
         del self.info.settings.build_type
 
+    def validate(self):
+        if self.settings.os not in ["Linux", "FreeBSD", "Windows", "Macos"] or self.settings.arch != "x86_64":
+            raise ConanInvalidConfiguration(f"{self.name} doesn't support current environment")
+
     def source(self):
```

Comment:
```suggestion
    def build(self):
```
When packaging pre-built binaries, the common senses is downloading to the build folder directly, it's can be removed after building as it does not contain sources: 

https://docs.conan.io/2/tutorial/creating_packages/other_types_of_packages/package_prebuilt_binaries.html#downloading-and-packaging-pre-built-binaries